articles/tutorial-qdk-quantum-random-number-generator.md

```diff
@@ -56,22 +70,69 @@ We can use this representation to visualize what the code is doing:
 
 Since the outcome of the measurement is completely random, we have obtained a random bit. We can call this operation several times to create integers. For example, if we call the operation three times to obtain three random bits, we can build random 3-bit numbers (that is, a random number between 0 and 7).
 
-
 ## Creating a complete random number generator
 
 Now that we have a Q# operation that generates random bits, we can use it to build a complete quantum random number generator. We can use a Q# application or use a host program.
 
-
-
 ### [Q# applications with Visual Studio or Visual Studio Code](#tab/tabid-qsharp)
 
 To create the full Q# application, add the following entry point to your Q# program: 
 
-:::code language="qsharp" source="~/quantum/samples/getting-started/qrng/Qrng.qs" range="17-33":::
+```qsharp
+operation SampleRandomNumberInRange(max : Int) : Int {
+    mutable bits = new Result[0];
+    for idxBit in 1..BitSizeI(max) {
+        set bits += [SampleQuantumRandomNumberGenerator()];
+    }
+    let sample = ResultArrayAsInt(bits);
+    return sample > max
+            ? SampleRandomNumberInRange(max)
+            | sample;
+}
+
+@EntryPoint()
+operation SampleRandomNumber() : Int {
+    let max = 50;
+    Message($"Sampling a random number between 0 and {max}: ");
+    return SampleRandomNumberInRange(max);
+}
+```
```

Comment:
```suggestion
```qsharp
    operation SampleRandomNumberInRange(max : Int) : Int {
        mutable bits = new Result[0];
        for idxBit in 1..BitSizeI(max) {
            set bits += [SampleQuantumRandomNumberGenerator()];
        }
        let sample = ResultArrayAsInt(bits);
        return sample > max
                ? SampleRandomNumberInRange(max)
                | sample;
    }
    
    @EntryPoint()
    operation SampleRandomNumber() : Int {
        let max = 50;
        Message($"Sampling a random number between 0 and {max}: ");
        return SampleRandomNumberInRange(max);
    }
```
```